### PR TITLE
Statically set woodwork typing in tests

### LINKED ISSUE #3651: Audit the Unit Test Suite for Woodwork Inference Sensitivity
In https://github.com/alteryx/evalml/pull/3626 and the much older https://github.com/alteryx/evalml/pull/2181, we've shown that EvalML is very sensitive to changes in Woodwork inference.  This was a mistake in how we've written our tests and we need to determine a way to rectify this.

I think resolving this story involves coming up with a plan to 👍 

1. Evaluate the changes in the linked PRs
2. Statically set woodwork typing in those tests/fixtures

### PR DESCRIPTION
**Pull request by eccabay**
Closes #3651 

This is not a woodwork 0.17.2 upgrade. The goal was simply to add explicit woodwork typing throughout the tests to make them more resilient to woodwork inference changes. The end result of this does have fewer tests failing in 0.17.2 since I used the upgrade as a check for general stability.
This does include some setting of woodwork types in components though, related to my previous work about being explicit about woodwork typing to reduce the number of times we re-infer within components (plus it makes tests pass 😁).

Changes in this PR:
- Consolidation of the `ts_data` and `get_ts_X_y` test pytest fixtures into one fixture, named `ts_data` but with data from `get_ts_X_y`. 
- Updated the fixtures `ts_data`, `imputer_test_data`, `X_y_binary`, `X_y_multi`, `X_y_regression`, `X_y_categorical_classification`, and `X_y_categorical_regression` to explicitly set woodwork types.
- Updated time series featurizer and data check test files to explicitly set woodwork types.
- `downcast_nullable_types` now works with either a DataFrame or a Series as input data (instead of just dataframes), and there's now a test for it.
- Removed a call to `ww.init()` in `infer_feature_types` in the case where we already have a valid schema for the input data. This has performance implications and there are performance test results for this branch.

Before vs after woodwork upgrade comparison:
Main/0.16.4 -> 0 failing tests
Main/0.17.2 -> 287 failing tests
Main /0.18.0 -> 291 failing tests

Branch/0.16.4 -> 0 failing tests
Branch/0.17.2 -> 23 failing tests
Branch/0.18.0 -> 22 failing tests

I apologize in advance for the length of this PR 😬

### CODE REVIEW

File: evalml/pipelines/components/transformers/imputers/time_series_imputer.py [hunk 1]
```diff
@@ -165,6 +165,11 @@ def transform(self, X, y=None):
 
         if self._interpolate_cols is not None:
             X_interpolate = X.ww[self._interpolate_cols]
+            # Pandas' interpolate function doesn't work with IntegerNullable types at this time
```
Comment (gsheni) [inline evalml/pipelines/components/transformers/imputers/time_series_imputer.py:168]: Let's put the link of the pandas issue to track this:
- https://github.com/pandas-dev/pandas/issues/40252

Comment (gsheni) [inline evalml/pipelines/components/transformers/imputers/time_series_imputer.py:168]: Let's put the link of the pandas issue in the comment
```suggestion
            # Pandas' interpolate function doesn't work with IntegerNullable types at this time
            # https://github.com/pandas-dev/pandas/issues/40252
```

**Comment by chukarsten** [comment]
Moving back to draft

File: evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py [hunk 2]
```diff
@@ -258,7 +260,7 @@ def _compute_delays(self, X_ww, y):
         # Features created from categorical columns should no longer be categorical
         lagged_features = pd.DataFrame(lagged_features)
         lagged_features.ww.init(
-            logical_types={col: "Double" for col in cols_derived_from_categoricals},
+            logical_types={col: "Double" for col in lagged_features.columns},
```
Comment (eccabay) [inline evalml/pipelines/components/transformers/preprocessing/time_series_featurizer.py:263]: I'm not confident in the correctness of this change, as to how we handle all types and if everything does in fact become a double here. If anyone knows better about this, please let me know.

File: evalml/tests/component_tests/test_simple_imputer.py [hunk 3]
```diff
@@ -568,7 +566,7 @@ def test_simple_imputer_ignores_natural_language(
             ans = X_df.mode().iloc[0, :]
             ans["natural language col"] = pd.NA
             X_df.iloc[-1, :] = ans
-        assert_frame_equal(result, X_df)
+        assert_frame_equal(result, X_df, check_dtype=False)
```
Comment (chukarsten) [inline evalml/tests/component_tests/test_simple_imputer.py:569]: Classic

**Review by chukarsten — APPROVED** [review]
Nice job Becca.  Really huge amount of work and very impressive that you lowered the sensitivity of the codebase to inference by so much!!

File: evalml/tests/utils_tests/test_woodwork_utils.py [hunk 4]
```diff
@@ -297,7 +297,22 @@ def test_schema_is_equal_fraud(fraud_100):
     assert _schema_is_equal(X.ww.schema, X2.ww.schema)
 
 
-def test_test_downcast_nullable_types_can_handle_no_schema():
+def test_downcast_nullable_types_series():
```
Comment (jeremyliweishih) [inline evalml/tests/utils_tests/test_woodwork_utils.py:300]: do you want to add the test case with `NaN`s as well?

File: evalml/utils/woodwork_utils.py [hunk 5]
```diff
@@ -118,42 +117,47 @@ def _schema_is_equal(first, other):
...
+        data: DataFrame or Series initialized with logical type information where BooleanNullable are cast as Double.
     """
-    if X.ww.schema is None:
-        X.ww.init()
+    if data.ww.schema is None:
+        data.ww.init()
 
-    X_bool_nullable_cols = X.ww.select("BooleanNullable")
-    X_int_nullable_cols = X.ww.select(["IntegerNullable", "AgeNullable"])
+    if isinstance(data, pd.Series):
```
Comment (jeremyliweishih) [inline evalml/utils/woodwork_utils.py:133]: if the series contains null values should we still downcast into Boolean or Double? I think we should follow what we have and use `ignore_null_cols` to differentiate.

**Review by jeremyliweishih — APPROVED** [review]
Great work on making the tests less reliant on inference. Just had a comment about nan behavior and down casting for series. Lmk what you think!

**Pull request merged**